iib/web/api_v1.py

```diff
@@ -168,24 +192,43 @@ def get_build_logs(request_id):
     :rtype: flask.Response
     :raise NotFound: if the request is not found or there are no logs for the request
     :raise Gone: if the logs for the build request have been removed due to expiration
+    :raise ValidationError: if the request has not completed yet
     """
     request_log_dir = flask.current_app.config['IIB_REQUEST_LOGS_DIR']
-    if not request_log_dir:
+    s3_bucket_name = flask.current_app.config['IIB_AWS_S3_BUCKET_NAME']
+    if not s3_bucket_name and not request_log_dir:
         raise NotFound()
```

Comment:
I know that you did not change it but is this really NotFound kind of error?  
For me it is more like the error you are raising below:
```
raise IIBError(
      'IIB is done processing the request and could not find logs.'
      ' Please make sure either an S3 bucket is configured or the logs are'
      ' stored locally in a directory by specifying IIB_REQUEST_LOGS_DIR'
)
```
 
But I am fine with keeping current behavior.

Reply:
This is only ever executed when IIB is not configured to serve logs at all. In this case, it's expected that it will always return the same status code regardless of the status of the request. Since this configuration is optional, it shouldn't be a server error.

The argument for treating it as a client error is that the client made a request to an endpoint which is not supported by the IIB instance. Of course, the client has no way of knowing that beforehand. I do think 404 makes the most sense here.

Reply:
@lcarva Thanks, I agree.